backend/services/matchService.js

```diff
@@ -11,7 +12,21 @@ module.exports = {
         }
         return result;
     },
+
     getMatchesByStudentId: (studentId) => {
         return matchDAO.getMatchesByStudentId(studentId);
+    },
+    
+    deleteMatch: async (studentId, placementId) => {
+        return await matchDAO.deleteMatch(studentId, placementId);
+    },
+
+    getMatchesByPlacementId: async (placementId) => {        
+        let students = await matchDAO.getMatchesByPlacementId(placementId);
+        let studentProfiles = [];
+        for (let i = 0; i<students.length; i++) {
+            studentProfiles.push(await studentService.getStudentProfile(students[i].studentId));
```

Comment:
this is something we want to avoid. this makes a separate query to the database for each student profile, besides the first one to get their ids, so for 20 matches, we would make 21 calls to DB! That can be solved by selecting in the get matches by placement id all the info that you need, through joins, in the same way that we did it on the previous similar issue in placementDAO.getPlacementsForSkills. You can check there how we build the final object to be send to the service. If we succeed, we will only have 1 call to the db!!

Reply:
yes, I've already tried to do it with joins (actually was the first thing that I tried) but it returns the same student as many times as the skills he has. And I don't know how to manipulate a json like that in the frontend, this is why I recycled the get profile function